### PR TITLE
automation: add a flag to enable/disable jobs

### PR DESCRIPTION
**Pull request by dossantospat**
## Overview

Adds a flag to the automation plan to enable or disable jobs. The flag can be toggled in the UI in the automation panel.

## Related Issues

Fix zaproxy/zaproxy#8626.

## Checklist
- [ ] Update help
- [ ] Update changelog
- [ ] Run `./gradlew spotlessApply` for code formatting
- [ ] Write tests
- [ ] Check code coverage
- [ ] Sign-off commits
- [ ] Squash commits
- [ ] Use a descriptive title

For more details, please refer to the [developer rules and guidelines](https://www.zaproxy.org/docs/developer/dev-rules-and-guidelines/).

### CODE REVIEW

File: addOns/automation/src/main/java/org/zaproxy/addon/automation/AutomationJob.java [hunk 1]
```diff
@@ -123,6 +125,14 @@ public String getSummary() {
         return EMPTY_SUMMARY;
     }
 
+    public boolean isEnabled() {
+        return enabled;
+    }
+
+    public void setEnabled(boolean enabled) {
+        this.enabled = enabled;
+    }
```
Comment (kingthorin) [inline addOns/automation/src/main/java/org/zaproxy/addon/automation/AutomationJob.java:134]: Could use https://github.com/zaproxy/zaproxy/blob/main/zap/src/main/java/org/zaproxy/zap/utils/EnableableInterface.java ?

**Comment by dossantospat** [comment]
I have read the CLA Document and I hereby sign the CLA

File: addOns/automation/src/main/java/org/zaproxy/addon/automation/AutomationJob.java (hunk 1 above)
Comment (thc202) [reply addOns/automation/src/main/java/org/zaproxy/addon/automation/AutomationJob.java:134]: No benefit for now.

File: addOns/automation/src/main/java/org/zaproxy/addon/automation/AutomationJob.java [hunk 2]
```diff
@@ -123,6 +125,14 @@ public String getSummary() {
         return EMPTY_SUMMARY;
     }
 
+    public boolean isEnabled() {
+        return enabled;
+    }
+
+    public void setEnabled(boolean enabled) {
+        this.enabled = enabled;
```
Comment (thc202) [inline addOns/automation/src/main/java/org/zaproxy/addon/automation/AutomationJob.java:133]: Add `setChanged()`.

File: addOns/automation/src/main/java/org/zaproxy/addon/automation/jobs/JobData.java [hunk 3]
```diff
@@ -54,6 +57,14 @@ public void setName(String name) {
         this.job.setName(name);
     }
 
+    public boolean isEnabled() {
```
[the remark this answers is not in the document]
Comment (dossantospat) [reply addOns/automation/src/main/java/org/zaproxy/addon/automation/jobs/JobData.java:62]: Done

File: addOns/automation/src/test/java/org/zaproxy/addon/automation/jobs/ActiveScanJobUnitTest.java [hunk 4]
```diff
@@ -133,6 +133,7 @@ void shouldReturnDefaultFields() {
         // Then
         assertThat(job.getType(), is(equalTo("activeScan")));
         assertThat(job.getName(), is(equalTo("activeScan")));
+        assertThat(job.isEnabled(), is(equalTo(true)));
```
[the remark this answers is not in the document]
Comment (dossantospat) [reply addOns/automation/src/test/java/org/zaproxy/addon/automation/jobs/ActiveScanJobUnitTest.java]: Done

File: addOns/automation/src/main/java/org/zaproxy/addon/automation/ExtensionAutomation.java [hunk 5]
```diff
@@ -367,42 +367,52 @@ public AutomationProgress runPlan(AutomationPlan plan, boolean resetProgress) {
...
-            if (View.isInitialised()) {
-                timer = new Timer(1000, e -> getAutomationPanel().updateJob(job));
-                timer.start();
-            }
-            try {
-                job.runJob(env, progress);
-            } catch (Exception e) {
-                LOGGER.error(e.getMessage(), e);
-                progress.error(
+            if (job.isEnabled()) {
```
[the remark this answers is not in the document]
Comment (dossantospat) [reply addOns/automation/src/main/java/org/zaproxy/addon/automation/ExtensionAutomation.java]: Done

File: addOns/automation/src/main/java/org/zaproxy/addon/automation/AutomationJob.java (hunk 2 above)
Comment (dossantospat) [reply addOns/automation/src/main/java/org/zaproxy/addon/automation/AutomationJob.java:133]: I called `jobChanged()`  directly  in `PlanTreeTableModel:: setValueAt()`, not sure if this is the right way :

```java
    @Override
    public void setValueAt(Object value, Object node, int column) {
        if (column == ENABLED_INDEX) {
            DefaultMutableTreeNode treeNode = (DefaultMutableTreeNode) node;
            if (!treeNode.isRoot()) {
                Object obj = treeNode.getUserObject();
                if (obj instanceof AutomationJob) {
                    AutomationJob job = (AutomationJob) obj;
                    job.setEnabled(((Boolean) value).booleanValue());
                    jobChanged(job);
                }
            }
        }
    }
```

**Review by kingthorin — APPROVED** [review]
LGTM

**Review by thc202 — APPROVED** [review]

**Comment by thc202** [comment]
Thank you!

**Pull request merged**